third-party/realdds/include/realdds/dds-device-impl.h

```diff
@@ -206,117 +236,106 @@ class dds_device::impl
                                            << state );
                             break;
                         case topics::device::notification::msg_type::MOTION_STREAM_PROFILES:
-                            if( state_type::WAIT_FOR_PROFILES == state )
+                            if ( state_type::WAIT_FOR_PROFILES == state )
                             {
                                 LOG_INFO( "got MOTION_STREAM_PROFILES message" );
-                                auto motion_stream_profiles
-                                    = data.get< topics::device::notification::motion_stream_profiles_msg >();
-
-                                // TODO: Try to save the "emplace" profiles copy
-                                topics::device::notification::motion_stream_profiles_msg msg;
-                                msg.dds_sensor_index = motion_stream_profiles->dds_sensor_index;
-                                msg.num_of_profiles = motion_stream_profiles->num_of_profiles;
-                                for( int i = 0; i < motion_stream_profiles->num_of_profiles; ++i )
-                                    msg.profiles[i] = motion_stream_profiles->profiles[i];
-
-                                auto sensor_name_it
-                                    = _sensor_index_to_name.find( motion_stream_profiles->dds_sensor_index );
-                                if( sensor_name_it != _sensor_index_to_name.end() )
+
+                                auto profiles_msg = data.get< topics::device::notification::motion_stream_profiles_msg >();
+                                if ( profiles_msg == nullptr )
                                 {
-                                    _sensor_to_motion_profiles.emplace( motion_stream_profiles->dds_sensor_index,
-                                                                        std::move( msg ) );
-
-                                    if( _sensor_to_video_profiles.size() + _sensor_to_motion_profiles.size()
-                                        < _num_of_sensors )
-                                        state = state_type::WAIT_FOR_SENSOR_HEADER;
-                                    else
-                                        state = state_type::DONE;
+                                    throw std::runtime_error( "get< topics::device::notification::motion_stream_profiles_msg > return nullptr" );
+                                }
+
+                                for ( size_t i = 0; i < profiles_msg->num_of_profiles; ++i )
+                                {
+                                    auto profile = profiles_msg->profiles[i];
+                                    auto key = std::make_pair( profile.uid, profile.stream_index );
+                                    if ( !_streams[key] )
+                                    {
+                                        _streams[key] = std::make_shared<dds_motion_stream>( profile.type, profiles_msg->group_name );
```

Comment:
As we discussed, I'd argue that profiles are `const` once the stream is created, and therefore I'd pass them in the constructor (so constructed only after you've collected all the profiles)

Reply:
For now, leave as is